src/frontend/src/scheduler/distributed/query.rs

```diff
@@ -94,6 +94,9 @@ struct QueryRunner {
     epoch: u64,
     hummock_snapshot_manager: HummockSnapshotManagerRef,
     compute_client_pool: ComputeClientPoolRef,
+
+    // Used for cleaning up `QueryExecution` after execution.
+    query_execution_map: QueryExecutionMap,
```

Comment:
Maybe we can add an `end_query` method on `QueryManager`, so that we don't have to expose its implementation to other structs?

Reply:
I think passing either `QueryManager` or `QueryExecutionMap` to `QueryRunner` is fine.